foreshadow/foreshadow.py

```diff
@@ -311,6 +316,110 @@ def score(self, data_df, y_df=None, sample_weight=None):
         self._prepare_predict(data_df.columns)
         return self.pipeline.score(data_df, y_df, sample_weight)
 
+    def dict_serialize(self, deep=True):
+        """Serialize the init parameters of the foreshaow object.
+
+        Args:
+            deep (bool): If True, will return the parameters for this estimator
+                recursively
+
+        Returns:
+            dict: The initialization parameters of the foreshadow object.
+
+        """
+        params = self.get_params(deep)
+        selected_params = self.customize_serialization_params(params)
+        serialized = _make_serializable(
+            selected_params, serialize_args=self.serialize_params
+        )
+        serialized["estimator"] = self.__customize_serialized_estimator(
+            self.estimator
+        )
+        return serialized
+
+    @staticmethod
+    def __customize_serialized_estimator(estimator):
+        result = dict()
+        is_meta_estimator = False
+        if isinstance(estimator, MetaEstimator):
+            estimator = estimator.estimator
+            is_meta_estimator = True
+
+        serialized_estimator = estimator.get_params()
+        serialized_estimator["_class"] = (
+            estimator.__module__ + "." + type(estimator).__name__
+        )
+        serialized_estimator["_method"] = "dict"
+
+        if is_meta_estimator:
+            result["estimator"] = serialized_estimator
+            result["preprocessor"] = "_y_preparer"
```

Comment:
You are correct. This if statement is unnecessary. The only caveat is that when we deserialize the foreshadow object back, the estimator is not a MetaEstimator unless we call `.fit` on the foreshadow object. I'm considering moving the following lines to the `__init__` method of the Foreshadow class:

```
# setup MetaEstimator if y_preparer is passed in
if self.y_preparer is not None:
    self.estimator = MetaEstimator(self.estimator, self.y_preparer)
```

Reply:
This might be the best time to clean up the properties that were used. I don't think the properties is the best way of auto initializing those attributes and think they should all be moved to the init, similar to what you just proposed :)